### PR TITLE
Improve redirects

### PR DESCRIPTION
**Pull request by vy**

### CODE REVIEW

File: src/site/resources/.htaccess [hunk 1]
```diff
@@ -32,12 +32,14 @@ RewriteRule "^(.+)$" "$1.html"
 # From this point rules match against the URI to the real file, with "/log4j/2.x/" stripped.
 
 # The content moved between pages
+RewriteRule "^articles\.html$" "manual/index.html" [R=permanent]
 RewriteRule "^log4j-core/apidocs(.*)$" "javadoc/log4j-core$1" [R=permanent]
 RewriteRule "^manual/api-separation\.html$" "manual/api.html" [R=permanent]
-RewriteRule "^manual/scala-api\.html$" "/log4j/scala/index.html" [R=permanent]
+RewriteRule "^manual/scala-api\.html$" "/log4j/scala" [R=permanent]
```
Comment (ppkarwasz) [inline src/site/resources/.htaccess:38]:
```suggestion
RewriteRule "^manual/scala-api\.html$" "/log4j/scala/latest/index.html" [R=permanent]
```

**Review by ppkarwasz — APPROVED** [review]
LGTM

File: src/site/resources/.htaccess (hunk 1 above)
Comment (vy) [reply src/site/resources/.htaccess:38]: Sorry, I am against this. `/log4j/scala` is an external site and I don't want to have assumptions on its implementation. This approach is future-proof against the problems you're suggesting to fix, i.e., `/index.html` has been moved to `/latest/index.html`.

File: src/site/resources/.htaccess [hunk 2]
```diff
@@ -47,22 +49,22 @@ RewriteRule "^runtime-dependencies\.html$" "manual/installation.html" [R=permane
...
 RewriteRule "^log4j-docker\.html/index\.html$" "log4j-docker.html" [R=permanent]
 RewriteRule "^log4j-flume-ng\.html/index\.html$" "log4j-flume-ng.html" [R=permanent]
 RewriteRule "^log4j-iostreams\.html/index\.html$" "log4j-iostreams.html" [R=permanent]
 RewriteRule "^log4j-jakarta-web\.html/index\.html$" "log4j-jakarta-web.html" [R=permanent]
-RewriteRule "^log4j-jcl\.html/index\.html$" "log4j-jcl.html" [R=permanent]
-RewriteRule "^log4j-jmx-gui\.html/index\.html$" "log4j-jmx-gui.html" [R=permanent]
-RewriteRule "^log4j-jpl\.html/index\.html$" "log4j-jpl.html" [R=permanent]
-RewriteRule "^log4j-jul\.html/index\.html$" "log4j-jul.html" [R=permanent]
+RewriteRule "^log4j-jcl\.html/index\.html$" "manual/installation.html#impl-core-bridge-jcl" [R=permanent]
+RewriteRule "^log4j-jmx-gui\.html/index\.html$" "/log4j/jmx-gui" [R=permanent]
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/resources/.htaccess:61]: Sorry, [I don't think this is the right approach](https://github.com/apache/logging-log4j2/pull/2622#discussion_r1616810454).

**Pull request merged**